### PR TITLE
MDC recommendation Guest Config source code

### PR DESCRIPTION
**Pull request by Jyotsna-Anand**
- Created a new folder for MDC under 'VMEncryption'. 
- Added Python helper scripts and extension_shim handler to 'helper-scripts'.
- Added mof files required for DSC module generation and C++ logic for DSC resource module to 'LinuxEncryptionCompliance' folder. 

**Note**: The skeleton code for DSC resource module in C++ is auto-generated by codegen.cmd. Most files under 'LinuxEncryptionCompliance' were auto generated. I've added them to the repo for tracking purposes. Only the below listed files carry business logic for ADE/HBE and need to be reviewed: 
1. Python helper scripts
2. Mof files
3. LinuxEncryptionCompliance.cpp under 'LinuxEncryptionCompliance' folder.

### CODE REVIEW

**Review by bamuir — APPROVED** [review]
Looks good to me. Is there any way to generate tests for these different flows/scenarios? Would be curious to see it in action as well if possible.

File: VMEncryption/mdc-recommendation/helper-scripts/VerifyVMSupport.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    is_kubernetes_vm = False 
+
+    if (microsoft_aks_image_publisher in vm_image_publisher) or (vm_image_sku.startswith(aks_prefix)):
+            is_kubernetes_vm = True
+    elif (microsoft_databrick_image_publisher in vm_image_publisher):
+        is_kubernetes_vm = True                   
+
+    return is_kubernetes_vm
+
+def is_databricks_vm(imds_metadata):
```
[the remark this answers is not in the document]
Comment (bamuir) [reply VMEncryption/mdc-recommendation/helper-scripts/VerifyVMSupport.py:62]: answered myself for anyone interested https://azure.microsoft.com/en-us/pricing/details/databricks/#overview

**Comment by Jyotsna-Anand** [comment]
> 

Thanks for reviewing Bam. I tested the Python script for these changes individually through our regression tests (by deploying the Python script as a custom script extension on the VM). 

I've also done E2E scenario tests by building a test package of the C++ and Python files and deploying them on Linux VMs in our test subscription. Will ping you the link to those test results.

File: VMEncryption/mdc-recommendation/helper-scripts/SupportedSkus.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    "Standard_F8",
+    "Standard_G1",
+    "Standard_G2",
+    "Standard_G3",
+    "Standard_G4",
+    "Standard_G5"
+    ]
+
+""" list of Linux OS image skus supported by ADE """
+supported_ade_image_list = [
```
[the remark this answers is not in the document]
Comment (Jyotsna-Anand) [reply VMEncryption/mdc-recommendation/helper-scripts/SupportedSkus.py:97]: For MDC, we compare the image publisher details with what is being returned from IMDS metadata to determine supportability. The SupportedOS.json that we currently have in our extension seems a little more abstract. As a part of the extension refactoring of the Linux code for MDC, I'll look into checking if there is a way to combine them into one file while still having the information needed for IMDS checks.

**Pull request merged**